### PR TITLE
Rebase onto Istio 1.14.3

### PR DESCRIPTION
**Pull request by jewertow**

### CODE REVIEW

File: pilot/pkg/serviceregistry/kube/controller/controller.go [hunk 1]
```diff
@@ -144,6 +147,12 @@ type Options struct {
 
 	// If meshConfig.DiscoverySelectors are specified, the DiscoveryNamespacesFilter tracks the namespaces this controller watches.
 	DiscoveryNamespacesFilter filter.DiscoveryNamespacesFilter
+
+	// EnableCRDScan determines whether the controller will list all CRDs
+	// present in the cluster, and subsequently only create watches on those
+	// that are. If this is set to false, all CRDs defined in the schema must be
+	// present for istiod to function.
+	EnableCRDScan bool
```
Comment (luksa) [inline pilot/pkg/serviceregistry/kube/controller/controller.go:156]: This addition seems misplaced. It belongs to the commit `[no-cluster-rbac] MAISTRA-1895: Add option to enable/disable CRD scan (#260) (#546)`

File: bin/update_maistra_deps.sh [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -eux
+
+UPDATE_BRANCH=${UPDATE_BRANCH:-"maistra-2.2"}
```
Comment (luksa) [inline bin/update_maistra_deps.sh]: Should this be `maistra-2.3`?

**Review by luksa — COMMENTED** [review]
Generally, looks ok, but I'd squash a few more commits and move the scanCRDs thing to the correct commit.

For example, I'd squash:
- `OSSM-1430: CNI: Watch for modified files with a prefix` into `[cni] MAISTRA-2132 Support deployment of multiple plugin versions in Istio CNI`
- `OSSM-1650 Make sure initialSync and event loop behave the same (#551)` into `[ior] MAISTRA-1400 Add IOR to Pilot (#556)`
- and a few others.

File: pilot/pkg/serviceregistry/kube/controller/controller.go (hunk 1 above)
Comment (jewertow) [reply pilot/pkg/serviceregistry/kube/controller/controller.go:156]: Done

File: bin/update_maistra_deps.sh (hunk 2 above)
Comment (jewertow) [reply bin/update_maistra_deps.sh]: Yes. I created branch `maistra-2.3` in api repo and updated this script.

**Review by luksa — APPROVED** [review]

**Pull request merged**